core/dtn7/src/core/application_agent.rs

```diff
@@ -77,3 +84,13 @@ impl SimpleApplicationAgent {
         }
     }
 }
+
+/// Removes a bundle from the store if its destination is a singleton endpoint
+fn remove_singleton_bundle_from_store(bundle: &Bundle) {
```

Comment:
this function is not really application agent specific.
makes more sense to promote it to `store_remove_if_singleton_bundle` right next to `store_remove`.
it might also be useful for some routing algorithms such as static as there is no need to keep singelton bundle around if passed to the next hop router.